extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -418,13 +424,14 @@ export class PublishDatabaseDialog {
 					data: data
 				});
 
-				// add SQLCMD Variables table if it wasn't there before
-				if (Object.keys(this.project.sqlCmdVariables).length === 0) {
-					this.formBuilder?.insertFormItem({
-						title: constants.sqlCmdTableLabel,
-						component: <azdata.TableComponent>this.sqlCmdVariablesTable
-					},
-						6);
+				if (Object.keys(result.sqlCmdVariables).length) {
+					// add SQLCMD Variables table if it wasn't there before
+					if (Object.keys(this.project.sqlCmdVariables).length === 0) {
+						this.formBuilder?.insertFormItem(<azdata.FormComponent>this.sqlCmdVariablesFormComponent, 6);
+					}
```

Comment:
6?

Reply:
it's the index that the table is being inserted at